test/rulesets/Statistics/statistics.jl

```diff
@@ -1,11 +1,33 @@
 @testset "mean" begin
     rng = MersenneTwister(999)
     n = 9
-    rrule_test(mean, randn(rng), (abs2, nothing), (randn(rng, n), randn(rng, n)))
-    X = randn(rng, n, n)
-    y, dX = rrule(mean, X; dims=1)
-    ȳ = randn(rng, size(y))
-    X̄_ad = dX(ȳ)
-    X̄_fd = j′vp(central_fdm(5, 1), x->mean(x, dims=1), ȳ, X)
-    @test X̄_ad ≈ X̄_fd rtol=1e-9 atol=1e-9
+
+    @testset "Basic" begin
+        rrule_test(
+            mean,
+            randn(rng),
+            (randn(rng, n),
+            randn(rng, n))
+        )
+    end
+
+    @testset "with function arg" begin
+        rrule_test(
+            mean,
+            randn(rng),
+            (abs2, nothing),
+            (randn(rng, n),
+            randn(rng, n))
+        )
+    end
+
+    @testset "with dims kwargs" begin
+        X = randn(rng, n, n)
+        y, mean_pullback = rrule(mean, X; dims=1)
+        ȳ = randn(rng, size(y))
+        _, dX = mean_pullback(ȳ)
+        X̄_ad = extern(dX)
+        X̄_fd = j′vp(central_fdm(5, 1), x->mean(x, dims=1), ȳ, X)
```

Comment:
we define `_fdm` in `test_utls` for this

```suggestion
        X̄_fd = j′vp(_fdm, x->mean(x, dims=1), ȳ, X)
```